src/vulkan-renderer/wrapper/instance.cpp

```diff
@@ -90,6 +88,29 @@ Instance::Instance(const std::string &application_name, const std::string &engin
     spdlog::debug("Requested Vulkan API version: {}.{}.{}", VK_VERSION_MAJOR(m_vk_api_version),
                   VK_VERSION_MINOR(m_vk_api_version), VK_VERSION_PATCH(m_vk_api_version));
 
+    // This code will throw an exception if the required version of Vulkan API is not available on the system
+    // It will also throw an exception if the available version of Vulkan API could not be queried
+    {
+        std::uint32_t api_version = 0;
+        if (const auto result = vkEnumerateInstanceVersion(&api_version); result != VK_SUCCESS) {
+            spdlog::error("Error: vkEnumerateInstanceVersion returned {}!", vk_tools::as_string(result));
+            return;
+        }
+
+        if (VK_VERSION_MAJOR(api_version) < VK_VERSION_MAJOR(m_vk_api_version) ||
+            VK_VERSION_MINOR(api_version) < VK_VERSION_MINOR(m_vk_api_version) ||
+            VK_VERSION_PATCH(api_version) < VK_VERSION_PATCH(m_vk_api_version)) {
+            throw std::runtime_error(
```

Comment:
This code is a little hard to read, it might be nicer to `fmt::format` it first.

Reply:
Good idea, done